taskgraph/Task.py

```diff
@@ -1533,9 +1532,11 @@ def _filter_non_files(
     elif isinstance(base_value, dict):
         for key in base_value.keys():
             value = base_value[key]
+            scrubbed_dict = {}
             for filter_value in _filter_non_files(
                     value, keep_list, ignore_list, keep_directories):
-                yield (value, filter_value)
+                scrubbed_dict[key] = filter_value
+            yield scrubbed_dict

```

Comment:
Should this `yield` be inside the for loop? If it's not I think it will overwrite nested values. Passing in:
`{0: {'one': 0, 'two': 1, 'three': 2}, 1: {'one': 1, 'two': 2, 'three': 3}, 2: {'one': 2, 'two': 3, 'three': 4}}`
gives me a generator that produces:
```
{0: {'three': 2}}
{1: {'three': 3}}
{2: {'three': 4}}
```
Where as kicking it inside the loop gives me:
```
{0: {'one': 0}}
{0: {'two': 1}}
{0: {'three': 2}}
{1: {'one': 1}}
{1: {'two': 2}}
{1: {'three': 3}}
{2: {'one': 2}}
{2: {'two': 3}}
{2: {'three': 4}}
```

Reply:
Gosh thanks for spotting this, there was a lot more going on there even than just that.